store/src/state.rs

```diff
@@ -478,17 +479,21 @@ impl State {
 // UTILITIES
 // ================================================================================================
 
-/// Returns the nullifier's leaf value in the SMT by its block number.
-fn block_num_to_nullifier_value(block: BlockNumber) -> Word {
-    [Felt::from(block), Felt::ZERO, Felt::ZERO, Felt::ZERO]
-}
+struct NullifierValueEncoder;
```

Comment:
I was thinking more along the lines of having a `NullifierValue` struct

```rust
struct NullifierValue(Word);
```

With conversions to/from `BlockNum`, etc. Basically the store and block producer would use this type instead of `Word` for nullifier values.

Reply:
@plafer it is needed to store this value in `Smt` and it doesn't support anything but `Word`.

Reply:
Ah, good point. Created this issue to [resolve this](https://github.com/0xPolygonMiden/crypto/issues/288). Until that is implemented, I personally prefer just a function over this `NullifierValueEncoder` struct - so IMO we should just close this PR.

What do you think?

Reply:
@plafer yes, I agree with you, it will be good solution. I'll close this PR, thank you!